client/export_test.go

```diff
@@ -30,11 +30,9 @@ func (client *Client) SetDoer(d doer) {
 	client.doer = d
 }
 
-type DoFlags = doFlags
-
 // Do does do.
-func (client *Client) Do(method, path string, query url.Values, body io.Reader, v interface{}, flags DoFlags) (statusCode int, err error) {
-	return client.do(method, path, query, nil, body, v, flags)
+func (client *Client) Do(method, path string, query url.Values, body io.Reader, v interface{}, opts *doOptions) (statusCode int, err error) {
```

Comment:
we should export `doOptions` too so we can test Do logic with the various options specifically

Reply:
did this